commit d456e2e391134742c633b0029251f7ca06bb5204
Author: Ganael Laplanche <ganael.laplanche@martymac.org>
Date:   Wed Oct 7 21:25:29 2020 +0200

    Resolve VkDeviceMemory/nullptr type mismatch to fix build on FreeBSD i386

--- Source/Core/VideoBackends/Vulkan/VKTexture.cpp.orig	2020-09-30 21:42:30 UTC
+++ Source/Core/VideoBackends/Vulkan/VKTexture.cpp
@@ -122,7 +122,7 @@ std::unique_ptr<VKTexture> VKTexture::CreateAdopted(co
                                                     VkImageViewType view_type, VkImageLayout layout)
 {
   std::unique_ptr<VKTexture> texture = std::make_unique<VKTexture>(
-      tex_config, nullptr, image, layout, ComputeImageLayout::Undefined);
+      tex_config, VkDeviceMemory(VK_NULL_HANDLE), image, layout, ComputeImageLayout::Undefined);
   if (!texture->CreateView(view_type))
     return nullptr;
 
--- Source/Core/VideoBackends/Vulkan/VKTexture.h.orig	2020-09-30 21:42:30 UTC
+++ Source/Core/VideoBackends/Vulkan/VKTexture.h
@@ -54,7 +54,7 @@ class VKTexture final : public AbstractTexture (public
   VkImageView GetView() const { return m_view; }
   VkImageLayout GetLayout() const { return m_layout; }
   VkFormat GetVkFormat() const { return GetVkFormatForHostTextureFormat(m_config.format); }
-  bool IsAdopted() const { return m_device_memory != nullptr; }
+  bool IsAdopted() const { return m_device_memory != VkDeviceMemory(VK_NULL_HANDLE); }
 
   static std::unique_ptr<VKTexture> Create(const TextureConfig& tex_config);
   static std::unique_ptr<VKTexture>
